### ISSUE TITLE
 Negative dimension size caused by subtracting 5 from 1 for 'max_pooling2d_3/MaxPool' (op: 'MaxPool') with input shapes: [?,1,199,128].

### ISSUE THREAD

**Issue by xdwang0726** [rstudio/keras#526]
I add parameter data_format = "channel_last" for both conv2D and maxpooling2D layers, but it still return channel first, what's happened? My code is below:

from keras.layers import Dense, Input, Flatten
from keras.layers import Conv2D, MaxPooling2D, Embedding, Reshape, Concatenate, Dropout
from keras import optimizers
from keras.models import Model

filter_sizes = [2,4,8]
BATCH_SIZE = 10

sequence_input = Input(shape=(MAX_SEQUENCE_LENGTH,), dtype='int32')
embedded_sequences = embedding_layer(sequence_input)
drop_out = Dropout(0.25)(embedded_sequences)
reshape = Reshape((MAX_SEQUENCE_LENGTH, EMBEDDING_DIM, 1))(drop_out)

conv_0 = Conv2D(filters = 128, kernel_size=(MAX_SEQUENCE_LENGTH, filter_sizes[0]), activation='relu', data_format = 'channels_last')(reshape)
conv_1 = Conv2D(filters = 128, kernel_size=(MAX_SEQUENCE_LENGTH, filter_sizes[1]), activation='relu', data_format = 'channels_last')(reshape)
conv_2 = Conv2D(filters = 128, kernel_size=(MAX_SEQUENCE_LENGTH, filter_sizes[2]), activation='relu', data_format = 'channels_last')(reshape)

maxpool_0 = MaxPooling2D(pool_size=(5,1), strides=None, padding='valid', data_format = 'channels_last')(conv_0)
maxpool_1 = MaxPooling2D(pool_size=(5,1), strides=None, padding='valid', data_format = 'channels_last')(conv_1)
maxpool_2 = MaxPooling2D(pool_size=(5,1), strides=None, padding='valid', data_format = 'channels_last')(conv_2)

**Comment by skeydan** [comment]
`channels_last` should be the default.What do you have in your `keras.json` config file?
(See https://keras.io/backend/).

**Comment by xdwang0726** [comment]
@skeydan Hi, I have "backend": "tensorflow".

**Comment by skeydan** [comment]
hm okay, for me it is

```
{
    "floatx": "float32",
    "epsilon": 1e-07,
    "backend": "tensorflow",
    "image_data_format": "channels_last"
}
```

You might want to try putting that last line in yours, too, restarting R and testing again.
If then you still have problems, please provide a minimal reproducible example so we can test for ourselves.
